lib/internal/streams/destroy.js

```diff
@@ -52,23 +55,44 @@ function destroy(err, cb) {
     r.destroyed = true;
   }
 
-  this._destroy(err || null, (err) => {
+  // If still constructing then defer calling _destroy.
+  if (!s.constructed) {
+    if (err || cb) {
+      // Slow path. Allocating closure. This should very rarely happen.
+      // This path can only be taken if the stream is destroyed with
+      // an error or a callback before it has been constructed.
+      s[kDestroy] = (stream, er) => {
+        _destroy(stream, err || er, cb);
+      };
+    } else {
+      // Fast path. Not allocating closure.
```

Comment:
I think we should call `_destroy` anyway. Why is it not called?

Reply:
It gets called in the construct callback. See the other comment with "fast path".

Reply:
I could make this a lot simpler to follow by just adding a `destroyCallback` property on the stream object. But that would add a small memory overhead.